org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardSwitchBranch.java

```diff
@@ -0,0 +1,166 @@
+	public IRepository selRepoData;
+	private final String destination;
+	private PageChangeListener pageChangeListener;
+	AbapGitWizardPageRepositoryAndCredentials pageCredentials;
+	AbapGitWizardPageSwitchBranchAndPackage pageBranchAndPackage;
+
+	public AbapGitWizardSwitchBranch(IProject project, IRepository selRepo, String destination) {
+		this.project = project;
+		this.cloneData = new CloneData();
+		this.destination = destination;
+		this.selRepoData = selRepo;
+		this.cloneData.url = selRepo.getUrl();
+		this.cloneData.branch = selRepo.getBranchName();
+		getPackageAndRepoType();
+
+		setWindowTitle(Messages.AbapGitWizardSwitch_branch_wizard_title);
+		setNeedsProgressMonitor(true);
+		setDefaultPageImageDescriptor(
+				AbstractUIPlugin.imageDescriptorFromPlugin(AbapGitUIPlugin.PLUGIN_ID, "icons/wizban/abapGit_import_wizban.png")); //$NON-NLS-1$
+	}
+
+	private void getRepositoryAccessMode() {
+		//Get external repo info for repository type (public / private)
+		IExternalRepositoryInfoService externalRepoInfoService = RepositoryServiceFactory
+				.createExternalRepositoryInfoService(AbapGitWizardSwitchBranch.this.destination, null);
+		AbapGitWizardSwitchBranch.this.cloneData.externalRepoInfo = externalRepoInfoService
+				.getExternalRepositoryInfo(AbapGitWizardSwitchBranch.this.selRepoData.getUrl(), "", "", null); //$NON-NLS-1$ //$NON-NLS-2$
+	}
+
+	private void getPackageRef(String packageName, IProgressMonitor monitor) {
+		IAdtPackageServiceUI packageServiceUI = AdtPackageServiceUIFactory.getOrCreateAdtPackageServiceUI();
+		if (packageServiceUI.packageExists(AbapGitWizardSwitchBranch.this.destination, packageName, monitor)) {
+			List<IAdtObjectReference> packageRefs = packageServiceUI.find(AbapGitWizardSwitchBranch.this.destination, packageName, monitor);
+			AbapGitWizardSwitchBranch.this.cloneData.packageRef = packageRefs.stream().findFirst().orElse(null);
+		}
+	}
+
+	public boolean getPackageAndRepoType() {
+
+		try {
+			String packageName = AbapGitWizardSwitchBranch.this.selRepoData.getPackage();
+			PlatformUI.getWorkbench().getProgressService().run(true, true, new IRunnableWithProgress() {
+
+				@Override
+				public void run(IProgressMonitor monitor) throws InvocationTargetException, InterruptedException {
+					monitor.beginTask(Messages.AbapGitWizardPageBranchAndPackage_task_package_validation_message, IProgressMonitor.UNKNOWN);
+
+					// fetches wether the repository is PUBLIC or PRIVATE from external repo info
+					getRepositoryAccessMode();
+
+					// fetches associated package
+					getPackageRef(packageName, monitor);
+
+				}
+			});
+			// returns false in case of missing package reference
+			return this.cloneData.packageRef != null;
+		} catch (InterruptedException | InvocationTargetException e) {
+			Shell shell = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell();
+			MessageDialog.openError(shell, "Error", e.getMessage()); //$NON-NLS-1$
+			return false;
+		}
+
+	}
```

Comment:
The idea is to not handle any exception here.
All exceptions should be handled by the caller.

Reply:
```suggestion
	public void getPackageAndRepoType() throws PackageRefNotFoundException {
			String packageName = AbapGitWizardSwitchBranch.this.selRepoData.getPackage();
			// fetches wether the repository is PUBLIC or PRIVATE from external repo info
			getRepositoryAccessMode();
			// fetches associated package
			getPackageRef(packageName);
	}
```